Realm/Realm/Realm.cs

```diff
@@ -1414,7 +1414,22 @@ public void WriteCopy(RealmConfigurationBase config)
                 throw new NotSupportedException($"Changing the partition to synchronize on is not supported when writing a Realm copy. Original partition: {originalConfig.Partition}, passed partition: {copiedConfig.Partition}");
             }
 
-            SharedRealmHandle.WriteCopy(config.DatabasePath, config.EncryptionKey);
+            if (SyncSession == null)
+            {
+                if (config is SyncConfigurationBase syncConfig)
+                {
+                    SharedRealmHandle.ExortTo(syncConfig.CreateNativeConfiguration(), syncConfig.EncryptionKey, syncConfig.CreateNativeSyncConfiguration(), syncConfig.GetSchema());
+                }
+                else
+                {
+                    SharedRealmHandle.ExortTo(config.CreateNativeConfiguration(), config.EncryptionKey, null, null);
+                }
+            }
+            else
+            {
+                var copyIsSync = config is SyncConfigurationBase;
+                SharedRealmHandle.WriteCopy(config.DatabasePath, config.EncryptionKey, copyIsSync);
+            }
```

Comment:
We don't need all this separate handling. We should be able to delete the `shared_realm_write_copy` method and instead only use the `ExportTo` functionality.

Reply:
I ended up keeping the name `write_copy` to keep it consistent with the SDK API. 👍